src/Pages/ViewList.js

```diff
@@ -1,23 +1,66 @@
-import React from 'react';
+// import { doc } from 'prettier';
+import React, { useState } from 'react';
 import { useCollection } from 'react-firebase-hooks/firestore';
 import { db } from '../lib/firebase';
 
 const ViewList = ({ token }) => {
   // if (token === null) token = 'default';
 
+  //{someArray.map((val, i) => <input onChange={e => handleInput(e, i)}></input>)}
+  //create  <input> elements and specify their name attributes
+
   const [list, loading, error] = useCollection(db.collection(token));
 
+  //const [filter, setFilter] =useState;
+  const [inputValue, setInputValue] = useState('');
+
+  const handleChange = (e) => {
+    e.preventDefault();
+    setInputValue(e.target.value);
+  };
+
   return (
     <div>
       <h2>Grocery List</h2>
+      <div>
+        <form action="#">
+          <label htmlFor="filter-list" style={{ display: 'hidden' }}>
+            Filter List
+          </label>
+          <br />
+          <input
+            type="search"
+            id="filter-list"
+            value={inputValue}
+            onChange={handleChange}
+            placeholder="Enter your item"
+          />
+        </form>
+      </div>
 
       {error && <strong>Error: {JSON.stringify(error)}</strong>}
       {loading && <span>Collection: Loading...</span>}
-      {list && (
+      {!loading && list && !inputValue && (
```

Comment:
We'll probably want to check for no errors here too 
```suggestion
      {!loading && !error && list && !inputValue && (
```